sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -155,7 +155,7 @@ object HiveAnalysis extends Rule[LogicalPlan] {
       CreateTableCommand(tableDesc, ignoreIfExists = mode == SaveMode.Ignore)
 
     case CreateTable(tableDesc, mode, Some(query)) if DDLUtils.isHiveTable(tableDesc) =>
-      DDLUtils.checkDataColNames(tableDesc)
+      DDLUtils.checkDataColNames(tableDesc.copy(schema = query.schema))
```

Comment:
How is it done for data source tables? by another rule?

Reply:
Yes in datasource table scenario, the flow will go through DataSourceAnalysis rule.

Reply:
And moreover one more problem what i observed is in serde class name defined "parquet.hive.serde.ParquetHiveSerDe" in checkDataColNames() API,  the serde name shall be "org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe" for parquet, so i added this serde also in above code as part of checkDataColNames()  API.

Reply:
can we unify this check for both data source table and hive serde table?

Reply:
sure, let me check. thanks for your input.

Reply:
Both are called from different rules, i will check how to unify

Reply:
Handled as part of PreprocessTableCreation rule for CTAS query, please review and let me know for any suggestions. Thanks